core/frontend/src/render/webgl/glsl/AthmosphericScattering.ts

```diff
@@ -0,0 +1,551 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import {
+  FragmentShaderComponent,
+  ProgramBuilder,
+  ShaderType,
+  VariablePrecision,
+  VariableType,
+} from "../ShaderBuilder";
+import { MAX_SAMPLE_POINTS } from "../AtmosphericScatteringUniforms";
+
+/** A physics-based atmospheric scattering technique that simulates how an atmosphere diverts light.
+ * @internal
+ * This shader adds an atmospheric scattering effect that mimics some aspects of the physical phenomenons of Rayleigh Scattering and Mie Scattering.
+ *
+ * This implementation is highly inspired by Sebastian Lague's Solar System project: https://github.com/SebLague/Solar-System/ and video: https://www.youtube.com/watch?v=DxfEbulyFcY
```

Comment:
Put these links and description in your main API documentation - no one will see them here.